privacyidea/lib/challenge.py

```diff
@@ -150,3 +151,21 @@ def _create_challenge_query(serial=None, transaction_id=None):
             sql_query = sql_query.filter(Challenge.transaction_id == transaction_id)
 
     return sql_query
+
+
+def extract_answered_challenges(challenges):
```

Comment:
See my comment above. Shouldn't we skip this method?

Reply:
I think as we want to keep logging serials of non-answered challenges to the audit log, we should keep this function to save one database query.

Reply:
ok.